### PR TITLE
feat: remove next dependency

### PR DESCRIPTION
**Pull request by chrishiguto**
### About

This PR removes Next.js dependency so our components can be compatible with any React framework.
Another piece of this PR is to add the first version of the Rockets Provider and the Rockets Navigation.

### Rockets Provider

Is a component that implements ClientProvider, AuthProvider and ThemeProvider making it easy all logic related to admin CRUDs in a single component. We also export very simple first version of a createConfig util function to generate the configs in a single object.

### Rockets Navigation

Is a plugin to be used together with Rockets Provider, this way the user is not tied to using react-router and, instead, can use any framework/library they prefer.

### CODE REVIEW

**Comment by MrMaz** [comment]
Just a reminder not to merge this until we have tagged 1.0.0 stable

File: packages/react-navigation/src/components/LoginRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { useAuth } from "@concepta/react-auth-provider";
+import { AuthModule } from "@concepta/react-material-ui/";
+
+const LoginRoute = () => {
+  const { accessToken: authAccessToken } = useAuth();
+
+  const accessToken = authAccessToken ?? localStorage.getItem("accessToken");
+
+  if (accessToken) {
+    return <Navigate to="/dealers" replace />;
```
Comment (andreneto97) [inline packages/react-navigation/src/components/LoginRoute.tsx:11]: dealers?

File: packages/react-navigation/src/components/LoginRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  return <AuthModule route="signIn" />;
```
Comment (andreneto97) [inline packages/react-navigation/src/components/LoginRoute.tsx:14]: These routes shoud be dynamic

File: packages/react-navigation/src/components/RoutesRoot.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    menuItems: DrawerItemProps[],
+    children: ReactNode
+  ) => ReactNode;
+};
+
+const RoutesRoot = ({ routes, items, renderAppBar }: RoutesRootProps) => {
+  return (
+    <Routes>
+      <Route path="/" element={<Navigate to={routes[0].props.id} replace />} />
+      <Route path="/login" element={<LoginRoute />} />
```
Comment (andreneto97) [inline packages/react-navigation/src/components/RoutesRoot.tsx:20]: this route should be dynamic

**Pull request merged**